Non passing tests for model with arrays and order by throw an exception

```shell
sqlmesh --version
0.131.0
```

When the test of the following model does not pass, an exception is thrown. When the `ORDER BY` part of the sql is removed, the exception is not thrown.

Model:
```sql
MODEL (
  name sandbox.order_by_and_arrays_bug,
  kind FULL
);

SELECT
    ARRAY_AGG(DISTINCT id_contact_b ORDER BY id_contact_b) AS aggregated_duplicates
FROM
    sandbox.x420_source
GROUP BY
    id_contact_a
ORDER BY
    id_contact_a
```

Passing test:
```yaml
test_arrays_bug:
  model: sandbox.order_by_and_arrays_bug
  inputs:
    sandbox.x420_source:
    - id_contact_a: a
      id_contact_b: b
    - id_contact_a: a
      id_contact_b: c
  outputs:
    query:
    - aggregated_duplicates:
      - b
      - c
```

Non passing test (b and c are in reverse order):
```yaml
test_arrays_bug:
  model: sandbox.order_by_and_arrays_bug
  inputs:
    sandbox.x420_source:
    - id_contact_a: a
      id_contact_b: b
    - id_contact_a: a
      id_contact_b: c
  outputs:
    query:
    - aggregated_duplicates:
      - c
      - b
```


Exception thrown when running test:
```shell
sqlmesh test
...E......
======================================================================
ERROR: test_arrays_bug (/project-dir/tests/test_arrays_bug.yaml)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/project-dir/.venv/lib/python3.12/site-packages/sqlmesh/core/test/definition.py", line 246, in assert_equal
    pd.testing.assert_frame_equal(
  File "/project-dir/.venv/lib/python3.12/site-packages/pandas/_testing/asserters.py", line 1279, in assert_frame_equal
    assert_series_equal(
  File "/project-dir/.venv/lib/python3.12/site-packages/pandas/_testing/asserters.py", line 1067, in assert_series_equal
    _testing.assert_almost_equal(
  File "testing.pyx", line 55, in pandas._libs.testing.assert_almost_equal
  File "testing.pyx", line 173, in pandas._libs.testing.assert_almost_equal
  File "/project-dir/.venv/lib/python3.12/site-packages/pandas/_testing/asserters.py", line 614, in raise_assert_detail
    raise AssertionError(msg)
AssertionError: DataFrame.iloc[:, 0] (column name="aggregated_duplicates") are different

DataFrame.iloc[:, 0] (column name="aggregated_duplicates") values are different (100.0 %)
[index]: [0]
[left]:  [[c, b]]
[right]: [[b, c]]
At positional index 0, first diff: ['c', 'b'] != ['b' 'c']

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/project-dir/.venv/lib/python3.12/site-packages/sqlmesh/core/test/definition.py", line 567, in runTest
    self.assert_equal(expected, actual, sort=sort, partial=partial)
  File "/project-dir/.venv/lib/python3.12/site-packages/sqlmesh/core/test/definition.py", line 269, in assert_equal
    diff = expected.compare(actual).rename(columns={"self": "exp", "other": "act"})
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project-dir/.venv/lib/python3.12/site-packages/pandas/core/frame.py", line 8593, in compare
    return super().compare(
           ^^^^^^^^^^^^^^^^
  File "/project-dir/.venv/lib/python3.12/site-packages/pandas/core/generic.py", line 10121, in compare
    mask = ~((self == other) | (self.isna() & other.isna()))
              ^^^^^^^^^^^^^
  File "/project-dir/.venv/lib/python3.12/site-packages/pandas/core/ops/common.py", line 76, in new_method
    return method(self, other)
           ^^^^^^^^^^^^^^^^^^^
  File "/project-dir/.venv/lib/python3.12/site-packages/pandas/core/arraylike.py", line 40, in __eq__
    return self._cmp_method(other, operator.eq)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project-dir/.venv/lib/python3.12/site-packages/pandas/core/frame.py", line 7900, in _cmp_method
    new_data = self._dispatch_frame_op(other, op, axis=axis)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project-dir/.venv/lib/python3.12/site-packages/pandas/core/frame.py", line 7956, in _dispatch_frame_op
    bm = self._mgr.operate_blockwise(
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project-dir/.venv/lib/python3.12/site-packages/pandas/core/internals/managers.py", line 1511, in operate_blockwise
    return operate_blockwise(self, other, array_op)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project-dir/.venv/lib/python3.12/site-packages/pandas/core/internals/ops.py", line 65, in operate_blockwise
    res_values = array_op(lvals, rvals)
                 ^^^^^^^^^^^^^^^^^^^^^^
  File "/project-dir/.venv/lib/python3.12/site-packages/pandas/core/ops/array_ops.py", line 344, in comparison_op
    res_values = comp_method_OBJECT_ARRAY(op, lvalues, rvalues)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project-dir/.venv/lib/python3.12/site-packages/pandas/core/ops/array_ops.py", line 127, in comp_method_OBJECT_ARRAY
    result = libops.vec_compare(x.ravel(), y.ravel(), op)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "ops.pyx", line 173, in pandas._libs.ops.vec_compare
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

----------------------------------------------------------------------
Ran 10 tests in 0.162s

FAILED (errors=1)
```

Model without `ORDER BY`:
```sql
MODEL (
  name sandbox.order_by_and_arrays_bug,
  kind FULL
);

SELECT
    ARRAY_AGG(DISTINCT id_contact_b ORDER BY id_contact_b) AS aggregated_duplicates
FROM
    sandbox.x420_source
GROUP BY
    id_contact_a
```

Resulting non passing test:
```shell
sqlmesh test
...F......
======================================================================
FAIL: test_arrays_bug (/project-dir/tests/test_arrays_bug.yaml)
----------------------------------------------------------------------
AssertionError: Data mismatch (exp: expected, act: actual)

  aggregated_duplicates        
                    exp     act
0                (c, b)  (b, c)

----------------------------------------------------------------------
Ran 10 tests in 0.159s

FAILED (failures=1)
```

--- a/sqlmesh/core/test/definition.py
+++ b/sqlmesh/core/test/definition.py
@@ -237,10 +237,11 @@ class ModelTest(unittest.TestCase):
                 return tuple((k, _to_hashable(v)) for k, v in x.items())
             return str(x) if not isinstance(x, t.Hashable) else x
 
+        actual = actual.apply(lambda col: col.map(_to_hashable))
+        expected = expected.apply(lambda col: col.map(_to_hashable))
+
         if sort:
-            actual = actual.apply(lambda col: col.map(_to_hashable))
             actual = actual.sort_values(by=actual.columns.to_list()).reset_index(drop=True)
-            expected = expected.apply(lambda col: col.map(_to_hashable))
             expected = expected.sort_values(by=expected.columns.to_list()).reset_index(drop=True)
 
         try:

--- a/sqlmesh/core/test/result.py
+++ b/sqlmesh/core/test/result.py
@@ -4,6 +4,12 @@ import types
 import typing as t
 import unittest
 
+if t.TYPE_CHECKING:
+    ErrorType = t.Union[
+        t.Tuple[type[BaseException], BaseException, types.TracebackType],
+        t.Tuple[None, None, None],
+    ]
+
 
 class ModelTextTestResult(unittest.TextTestResult):
     successes: t.List[unittest.TestCase]
@@ -12,13 +18,28 @@ class ModelTextTestResult(unittest.TextTestResult):
         super().__init__(*args, **kwargs)
         self.successes = []
 
-    def addFailure(
+    def addSubTest(
         self,
         test: unittest.TestCase,
-        err: (
-            tuple[type[BaseException], BaseException, types.TracebackType] | tuple[None, None, None]
-        ),
+        subtest: unittest.TestCase,
+        err: t.Optional[ErrorType],
     ) -> None:
+        """Called at the end of a subtest.
+
+        The traceback is suppressed because it is redundant and not useful.
+
+        Args:
+            test: The test case.
+            subtest: The subtest instance.
+            err: A tuple of the form returned by sys.exc_info(), i.e., (type, value, traceback).
+        """
+        if err:
+            exctype, value, tb = err
+            err = (exctype, value, None)  # type: ignore
+
+        super().addSubTest(test, subtest, err)
+
+    def addFailure(self, test: unittest.TestCase, err: ErrorType) -> None:
         """Called when the test case test signals a failure.
 
         The traceback is suppressed because it is redundant and not useful.

--- a/tests/core/test_test.py
+++ b/tests/core/test_test.py
@@ -388,7 +388,6 @@ test_array_order:
     )
 
 
-
 @pytest.mark.parametrize(
     "waiter_names_input",
     [

--- a/tests/core/test_test.py
+++ b/tests/core/test_test.py
@@ -345,6 +345,49 @@ test_foo:
         ),
     )
 
+    model_sql = """
+SELECT
+    ARRAY_AGG(DISTINCT id_contact_b ORDER BY id_contact_b) AS aggregated_duplicates
+FROM
+    source
+GROUP BY
+    id_contact_a
+ORDER BY
+    id_contact_a
+    """
+
+    _check_successful_or_raise(
+        _create_test(
+            body=load_yaml(
+                """
+test_array_order:
+  model: test
+  inputs:
+    source:
+    - id_contact_a: a
+      id_contact_b: b
+    - id_contact_a: a
+      id_contact_b: c
+  outputs:
+    query:
+    - aggregated_duplicates:
+      - c
+      - b
+                """
+            ),
+            test_name="test_array_order",
+            model=_create_model(model_sql),
+            context=Context(config=Config(model_defaults=ModelDefaultsConfig(dialect="duckdb"))),
+        ).run(),
+        expected_msg=(
+            """AssertionError: Data mismatch (exp: expected, act: actual)\n\n"""
+            "  aggregated_duplicates        \n"
+            "                    exp     act\n"
+            "0                (c, b)  (b, c)\n"
+        ),
+    )
+
+
 
 @pytest.mark.parametrize(
     "waiter_names_input",

